safesocket/safesocket_darwin_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package safesocket
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestDarwinSetCredentials(t *testing.T) {
+	expectedPort := 123
+	expectedToken := "token"
+	SetCredentials(expectedToken, expectedPort)
+
+	port, token, err := LocalTCPPortAndToken()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if port != expectedPort {
+		t.Errorf("expected port %d, got %d", expectedPort, port)
+	}
+
+	if token != expectedToken {
+		t.Errorf("expected token %s, got %s", expectedToken, token)
+	}
+}
+
+func TestInitListenerDarwin(t *testing.T) {
+	temp := t.TempDir()
+	ln, err := InitListenerDarwin(temp)
+	if err != nil || ln == nil {
+		t.Fatalf("InitListenerDarwin failed: %v", err)
+	}
+	defer (*ln).Close()
+
+	port, token, err := LocalTCPPortAndToken()
+	if err != nil {
+		t.Fatalf("LocalTCPPortAndToken failed: %v", err)
+	}
+
+	if port == 0 {
+		t.Errorf("expected non-zero port, got %d", port)
+	}
+
+	if token == "" {
+		t.Errorf("expected non-empty token, got empty string")
+	}
+
+	// It's a hex string, so 2x the byte length
+	if len(token) != sameUserProofTokenLength*2 {
+		t.Errorf("expected token length %d, got %d", sameUserProofTokenLength, len(token))
+	}
+
+	// A token should only ever be generated once
+	nextToken, _ := getToken()
+	if nextToken != token {
+		t.Errorf("expected same token, got different token")
+	}
+}
+
+func TestMacsysSameuserproof(t *testing.T) {
+	isMacSysExt = true
+	verifyConn = false
+	dir := t.TempDir()
+	wantToken := "token"
+	wantPort := 123
+
+	initSameUserProofToken(dir, wantPort, wantToken)
+
+	gotPort, gotToken, err := readMacsysSameUserProof(dir)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if gotPort != wantPort {
+		t.Errorf("expected port %d, got %d", wantPort, gotPort)
```

Comment:
nit: Go test style is got first, then want.   (and generally uses the word "want" instead of "expected")